src/main/kotlin/org/rust/ide/typing/paste/RsImportCopyPasteProcessor.kt

```diff
@@ -94,12 +151,12 @@ class RsImportCopyPasteProcessor : CopyPastePostProcessor<RsTextBlockTransferabl
 
         val data = values.getOrNull(0) ?: return
         val file = editor.document.toPsiFile(project) as? RsFile ?: return
-        val range = bounds.range
 
-        val elements = gatherElements(file, range)
+        val elements = gatherElements(file, bounds.range)
         val importCtx = elements.firstOrNull { it is RsElement } as? RsElement ?: return
 
-        val visitor = ImportingVisitor(range, data.importMap)
+        val importOffset = bounds.range.startOffset
+        val visitor = ImportingVisitor(importOffset, data.importMap)
 
         runWriteAction {
             for (element in elements) {
```

Comment:
Btw turns out elements contains not only top-level items, so we run auto import multiple times for same path, which may hurt performance. I suggest either remove recursion from the visitors, or maybe even refactor this with simple loop and `is RsPath`, `is RsMethod` checks without visitor at all

![image](https://user-images.githubusercontent.com/6505554/196043206-1c097a4b-d0e6-42ca-8700-1794f4bcb94b.png)